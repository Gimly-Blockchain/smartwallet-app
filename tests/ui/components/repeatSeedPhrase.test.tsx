import React from 'react'
import { shallow } from 'enzyme'
import RepeatSeedPhraseComponent from '../../../src/ui/recovery/components/repeatSeedPhrase'

describe('repeatSeedPhrase component', () => {
  const makeProps = (otherProps: {}) => ({
    note: 'some note to let the user know what to do',
    mnemonicSorting: {},
    randomWords: ['Mock', 'Seed', 'Phrase'],
    back: jest.fn(),
    checkMnemonic: jest.fn(),
    selectPosition: jest.fn(),
    ...otherProps,
  })

  it('matches the snapshot without sorting', () => {
    const props = makeProps({})
    const rendered = shallow(<RepeatSeedPhraseComponent {...props} />)
    expect(rendered).toMatchSnapshot()
  })

  it('matches the snapshot with sorting', () => {
    const props = makeProps({
      mnemonicSorting: { 1: 'Mock', 3: 'Seed', 9: 'Phrase' },
      randomWords: [],
    })
    const rendered = shallow(<RepeatSeedPhraseComponent {...props} />)
    expect(rendered).toMatchSnapshot()
  })
})
