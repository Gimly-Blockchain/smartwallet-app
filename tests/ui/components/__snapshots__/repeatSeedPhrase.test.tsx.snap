// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`repeatSeedPhrase component matches the snapshot with sorting 1`] = `
<Container
  style={
    Object {
      "backgroundColor": "black",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "marginTop": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffefdf",
            "fontFamily": "TT Commons",
            "fontSize": 20,
            "lineHeight": 26,
            "textAlign": "center",
          }
        }
      >
        some note to let the user know what to do
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 40,
          "marginTop": 15,
          "position": "relative",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "position": "absolute",
            },
            Object {
              "left": -0,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 18,
        }
      }
    >
      <View>
        <Placeholder
          i={0}
          key="0"
          onPress={[MockFunction]}
          sorting={
            Object {
              "1": "Mock",
              "3": "Seed",
              "9": "Phrase",
            }
          }
        />
        <Placeholder
          i={1}
          key="1"
          onPress={[MockFunction]}
          sorting={
            Object {
              "1": "Mock",
              "3": "Seed",
              "9": "Phrase",
            }
          }
        />
        <Placeholder
          i={2}
          key="2"
          onPress={[MockFunction]}
          sorting={
            Object {
              "1": "Mock",
              "3": "Seed",
              "9": "Phrase",
            }
          }
        />
        <Placeholder
          i={3}
          key="3"
          onPress={[MockFunction]}
          sorting={
            Object {
              "1": "Mock",
              "3": "Seed",
              "9": "Phrase",
            }
          }
        />
        <Placeholder
          i={4}
          key="4"
          onPress={[MockFunction]}
          sorting={
            Object {
              "1": "Mock",
              "3": "Seed",
              "9": "Phrase",
            }
          }
        />
        <Placeholder
          i={5}
          key="5"
          onPress={[MockFunction]}
          sorting={
            Object {
              "1": "Mock",
              "3": "Seed",
              "9": "Phrase",
            }
          }
        />
      </View>
      <View>
        <Placeholder
          i={6}
          key="0"
          onPress={[MockFunction]}
          sorting={
            Object {
              "1": "Mock",
              "3": "Seed",
              "9": "Phrase",
            }
          }
        />
        <Placeholder
          i={7}
          key="1"
          onPress={[MockFunction]}
          sorting={
            Object {
              "1": "Mock",
              "3": "Seed",
              "9": "Phrase",
            }
          }
        />
        <Placeholder
          i={8}
          key="2"
          onPress={[MockFunction]}
          sorting={
            Object {
              "1": "Mock",
              "3": "Seed",
              "9": "Phrase",
            }
          }
        />
        <Placeholder
          i={9}
          key="3"
          onPress={[MockFunction]}
          sorting={
            Object {
              "1": "Mock",
              "3": "Seed",
              "9": "Phrase",
            }
          }
        />
        <Placeholder
          i={10}
          key="4"
          onPress={[MockFunction]}
          sorting={
            Object {
              "1": "Mock",
              "3": "Seed",
              "9": "Phrase",
            }
          }
        />
        <Placeholder
          i={11}
          key="5"
          onPress={[MockFunction]}
          sorting={
            Object {
              "1": "Mock",
              "3": "Seed",
              "9": "Phrase",
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": "auto",
      }
    }
  >
    <ThemedComponent
      onPress={[MockFunction]}
      raised={true}
      style={
        Object {
          "container": Object {
            "backgroundColor": "#942f51",
            "borderRadius": 4,
            "height": 48,
            "paddingHorizontal": 25,
          },
          "text": Object {
            "color": "white",
            "fontFamily": "TT Commons",
            "fontSize": 22,
            "fontWeight": "100",
            "paddingVertical": 15,
          },
        }
      }
      text="[missing \\"en.Confirm and check\\" translation]"
      upperCase={false}
    />
  </View>
</Container>
`;

exports[`repeatSeedPhrase component matches the snapshot without sorting 1`] = `
<Container
  style={
    Object {
      "backgroundColor": "black",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "marginTop": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffefdf",
            "fontFamily": "TT Commons",
            "fontSize": 20,
            "lineHeight": 26,
            "textAlign": "center",
          }
        }
      >
        some note to let the user know what to do
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 40,
          "marginTop": 15,
          "position": "relative",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "position": "absolute",
            },
            Object {
              "left": -30,
            },
          ]
        }
      >
        <Text
          key="Mock"
          style={
            Array [
              Object {
                "color": "rgba(255, 239, 223, 0.38)",
                "fontFamily": "TT Commons",
                "fontSize": 34,
                "marginRight": 7,
              },
              Object {
                "alignSelf": "center",
                "color": "white",
                "position": "relative",
              },
            ]
          }
        >
          Mock
        </Text>
        <Text
          key="Seed"
          style={
            Array [
              Object {
                "color": "rgba(255, 239, 223, 0.38)",
                "fontFamily": "TT Commons",
                "fontSize": 34,
                "marginRight": 7,
              },
              false,
            ]
          }
        >
          Seed
        </Text>
        <Text
          key="Phrase"
          style={
            Array [
              Object {
                "color": "rgba(255, 239, 223, 0.38)",
                "fontFamily": "TT Commons",
                "fontSize": 34,
                "marginRight": 7,
              },
              false,
            ]
          }
        >
          Phrase
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 18,
        }
      }
    >
      <View>
        <Placeholder
          i={0}
          key="0"
          onPress={[MockFunction]}
          sorting={Object {}}
        />
        <Placeholder
          i={1}
          key="1"
          onPress={[MockFunction]}
          sorting={Object {}}
        />
        <Placeholder
          i={2}
          key="2"
          onPress={[MockFunction]}
          sorting={Object {}}
        />
        <Placeholder
          i={3}
          key="3"
          onPress={[MockFunction]}
          sorting={Object {}}
        />
        <Placeholder
          i={4}
          key="4"
          onPress={[MockFunction]}
          sorting={Object {}}
        />
        <Placeholder
          i={5}
          key="5"
          onPress={[MockFunction]}
          sorting={Object {}}
        />
      </View>
      <View>
        <Placeholder
          i={6}
          key="0"
          onPress={[MockFunction]}
          sorting={Object {}}
        />
        <Placeholder
          i={7}
          key="1"
          onPress={[MockFunction]}
          sorting={Object {}}
        />
        <Placeholder
          i={8}
          key="2"
          onPress={[MockFunction]}
          sorting={Object {}}
        />
        <Placeholder
          i={9}
          key="3"
          onPress={[MockFunction]}
          sorting={Object {}}
        />
        <Placeholder
          i={10}
          key="4"
          onPress={[MockFunction]}
          sorting={Object {}}
        />
        <Placeholder
          i={11}
          key="5"
          onPress={[MockFunction]}
          sorting={Object {}}
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": "auto",
      }
    }
  >
    <ThemedComponent
      onPress={[MockFunction]}
      raised={true}
      style={
        Object {
          "container": Object {
            "backgroundColor": "#942f51",
            "borderRadius": 4,
            "height": 48,
            "paddingHorizontal": 25,
          },
          "text": Object {
            "color": "white",
            "fontFamily": "TT Commons",
            "fontSize": 22,
            "fontWeight": "100",
            "paddingVertical": 15,
          },
        }
      }
      text="[missing \\"en.Show my phrase again\\" translation]"
      upperCase={false}
    />
  </View>
</Container>
`;
